Samples/CommandBinding/MauiEdit.ViewModels/Controller/MainFormController.cs

```diff
@@ -0,0 +1,237 @@
+﻿using System.Diagnostics;
+using CommunityToolkit.Mvvm.ComponentModel;
+
+namespace MauiEdit.ViewModels;
+
+/// <summary>
+///  Main ViewModel for the WinForms/MAUI-Editor sample app. Encapsulated the full logic of the app.
+/// </summary>
+public partial class MainFormController : ViewController
+{
+    // Holds the whole editor document.
+    private string? _textDocument;
+
+    // Backing field to iterate line-wise through the whole document.
+    private ReadOnlyMemory<char>[] _lines = Array.Empty<ReadOnlyMemory<char>>();
+
+    /// <summary>
+    ///  Gets the selection start- and end line based on SelectionIndex and SelectionLength.
+    ///  Annotated with <see cref="ObservablePropertyAttribute"/>, so the property is automatically code-generated.
+    /// </summary>
+    [ObservableProperty]
+    private (int StartLine, int EndLine) _selectionLines;
+
+    /// <summary>
+    ///  Current length of the selection.
+    /// </summary>
+    [ObservableProperty]
+    private int _selectionLength;
+
+    /// <summary>
+    ///  Column where the cursor is located.
+    /// </summary>
+    [ObservableProperty]
+    private int _selectionColumn;
+
+    /// <summary>
+    ///  Row where the cursor is located.
+    /// </summary>
+    [ObservableProperty]
+    private int _selectionRow;
+
+    /// <summary>
+    ///  The index of the first selected character 
+    ///  (or the current cursor position index in to the editor's doc).
+    /// </summary>
+    [ObservableProperty]
+    private int _selectionIndex;
+
+    /// <summary>
+    ///  Character count threshold until we wrap to the next line when calling the wrap command.
+    /// </summary>
+    [ObservableProperty]
+    private int _charCountWrapThreshold = 60;
+
+    /// <summary>
+    ///  Creates an instance of this class.
+    /// </summary>
+    /// <param name="serviceProvider"></param>
+    public MainFormController(IServiceProvider serviceProvider)
+        : base(serviceProvider)
+    {
+        // The initial sample text:
+        TextDocument = $"""
+                This is a test document.
+                It doesn't contain a lot of information.
+                But it contains enough, to test this app.
+
+                Just a few lines of text are sufficient for that.
+
+                Today is {DateTime.Now:dddd, MMMM dd. yyyy}. It is {DateTime.Now: hh:mm tt}.
+
+                The end.
+                """;
+    }
+
+    /// <summary>
+    ///  Provides the Document for the Editor.
+    /// </summary>
+    public string? TextDocument
+    {
+        get => _textDocument;
+
+        set
+        {
+            var wasEmpty = string.IsNullOrEmpty(_textDocument);
+
+            if (SetProperty(ref _textDocument, value))
+            {
+                _lines = GetLinesFromDocument();
+
+                if (wasEmpty ^ string.IsNullOrEmpty(_textDocument))
```

Comment:
This is hard to follow. Why exclusive or?

Reply:
Every time the status changed to the exact opposite (The document _was_ empty, but now it's not anymore, or it was _not_ empty, but now it is), we need to inform the respective commands, that they should re-query their execution status. That controls the enable state of the respective menu items.